coldfront/plugins/qumulo/management/commands/add_qumulo_project_attribute_types.py

```diff
@@ -0,0 +1,39 @@
+from coldfront.core.project.models import ProjectAttributeType, AttributeType
+from django.core.management.base import BaseCommand
+
+
+class Command(BaseCommand):
+    def handle(self, *args, **options):
+        print("Adding Qumulo Project Attribute Types")
+        # add a department_number at the project level that is *not* used for direct billing
+        # could maybe have it default for Allocation?
+        # NOTE - there is no UserAttributeType collection, so this is a more natural spot
+        ProjectAttributeType.objects.get_or_create(
+            attribute_type=AttributeType.objects.get(name="Text"),
+            name="sponsor_department_number",
+            is_required=True,
+            is_private=False,
+            is_unique=False,
+            is_changeable=True,
+        )
+
+        # NOTE: we'll keep "regular" allocations (even condo allocations) in "regular"
+        # projects; a condo_group project is a special thing that lets us check the
+        # sum of its allocations and bill them differently
+        ProjectAttributeType.objects.get_or_create(
+            attribute_type=AttributeType.objects.get(name="Yes/No"),
+            name="is_condo_group",
+            is_required=True,
+            is_private=False,
+            is_unique=False,
+            is_changeable=True,
+        )
+
+        ProjectAttributeType.objects.get_or_create(
+            attribute_type=AttributeType.objects.get(name="Int"),
```

Comment:
We should discuss this one.

There is code in `ris-itsm-reset` that will refuse to create an allocation if that service provision's quota is over a limit for the condo group.

See `src/jenkins.js` line 455:

`// If the service provision quota exceeds the maximum allocation for the condo group,
  // don't create the allocation.
  const parentId = data.parent_id;
  if (parentId) {
    // It should include condo group project allocations with status: active, pending, jenkins_error, and new.
    const condoAllocatedQuotasResult = await runSQL([
      "SELECT quota FROM billing.service_provision WHERE status in ('active', 'pending', 'jenkins_error', 'new') AND parent_id = ?",
      [parentId],
    ]);

    const condoMaximumQuotaResult = await runSQL([
      'SELECT quota FROM billing.service_provision WHERE id = ?',
      [parentId],
    ]);

    const condoAllocatedQuotas = condoAllocatedQuotasResult.rows.map((entry) => entry.quota);
    const condoMaximumQuota = condoMaximumQuotaResult.rows[0];

    if (parseQuotas(condoAllocatedQuotas) > parseQuota(condoMaximumQuota.quota)) {
      console.log(
        'Jenkins job was not a called due to overallocation of the condo. At the point of the request, any changes to the DB must have been persisted.'
      );
      return undefined;
    }
  }`

Reply:
We'll turn the `condo_group` cumulative quota limit into a fast-follow.

@peaches7083 to remove the new `quota_limit` `ProjectAttribute`.

Reply:
Handling this in another PR that will be merged into this branch.